spikeinterface/extractors/nwbextractors.py

```diff
@@ -0,0 +1,310 @@
+                        val = nwbfile.electrodes[col][electrode_table_index]
+                        if col not in properties:
+                            properties[col] = np.zeros(self.get_num_channels(), dtype=type(val))
+                        properties[col][es_ind] = val
+
+            for prop_name, values in properties.items():
+                if prop_name == "location":
+                    self.set_dummy_probe_from_locations(values)
+                elif prop_name == "group":
+                    self.set_channel_groups(val)
+                else:
+                    self.set_property(prop_name, values)
+
+            self._kwargs = {'file_path': str(Path(file_path).absolute()),
+                            'electrical_series_name': electrical_series_name}
+
+class NwbRecordingSegment(BaseRecordingSegment):
+    def __init__(self, path: PathType, electrical_series_name, num_frames):
+        BaseRecordingSegment.__init__(self)
+        self._path = path
+        self._electrical_series_name = electrical_series_name
+        self._num_samples = num_frames
+
+    def get_num_samples(self):
+        """Returns the number of samples in this signal block
+
+        Returns:
+            SampleIndex: Number of samples in the signal block
+        """
+        return self._num_samples
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        with NWBHDF5IO(self._path, 'r') as io:
+            nwbfile = io.read()
+            es = nwbfile.acquisition[self._electrical_series_name]
+
+            if isinstance(channel_indices, slice):
+                traces = es.data[start_frame:end_frame, channel_indices]
+            else:
+                # channel_indices is np.ndarray
+                if np.array(channel_indices).size > 1 and np.any(np.diff(channel_indices) < 0):
+                    # get around h5py constraint that it does not allow datasets
+                    # to be indexed out of order
+                    sorted_channel_indices = np.sort(channel_indices)
+                    resorted_indices = np.array([list(sorted_channel_indices).index(ch) for ch in channel_indices])
+                    recordings = es.data[start_frame:end_frame, sorted_channel_indices]
+                    traces = recordings[:, resorted_indices]
+                else:
+                    traces = es.data[start_frame:end_frame, channel_indices]
+                
+        return traces
+
+
+class NwbSortingExtractor(BaseSorting):
+    extractor_name = 'NwbSorting'
+    installed = HAVE_NWB  # check at class level if installed or not
+    is_writable = True
+    mode = 'file'
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+
+    def __init__(self, file_path, electrical_series: str = None, sampling_frequency: float = None):
```

Comment:
For consistency I would put electrical_series_name here.
It will help for the `read_nwb()`